uci.go

```diff
@@ -252,12 +255,16 @@ func (t *tree) Set(config, section, option string, values ...string) bool {
 	if opt := sec.Get(option); opt != nil {
 		opt.SetValues(values...)
 	} else {
-		sec.Add(newOption(option, values...))
+		sec.Add(newOption(option, optionType, values...))
 	}
 	cfg.tainted = true
 	return true
 }
 
+func (t *tree) Set(config, section, option string, values ...string) bool {
+	return t.SetType(config, section, option, TypeOption, values...)
+}
```

Comment:
Hm, maybe the `OptionType` here should depend on `len(values)`? This was previously decided in `types.go`:

```go
for _, opt := range sec.Options {
	switch l := len(opt.Values); {
	case l == 1:
		// option
	case l > 1:
		// list
	}
}
```

Additionally, I'm in favour of deprecating `Set`:

```suggestion
// Set will either set an option or a list at the given configuration section.
// The option/list decision is based on the number of values.
// Deprecated: Use SetType to control the type.
func (t *tree) Set(config, section, option string, values ...string) bool {
	if len(values) > 1 {
		return t.SetType(config, section, option, TypeList, values...)
	}
	return t.SetType(config, section, option, TypeOption, values...)
}
```

Reply:
I just noticed: the comment should go into the `Tree` interface.

Reply:
This was part of the logic that was causing a failure. It's totally valid for a list to only have 1 value. List and option are not interchangeable.